tests/drivers/can/api/testcase.yaml

```diff
@@ -18,3 +16,13 @@ tests:
       - esp32s2_saola
       - esp32s3_devkitm
       - xiao_esp32s3
+  drivers.can.api.nxp_s32_canxl_non_rx_fifo:
+    extra_configs:
+      - CONFIG_CAN_NXP_S32_RX_FIFO=n
+      - CONFIG_CAN_FD_MODE=n
```

Comment:
If `CONFIG_CAN_NXP_S32_RX_FIFO=n` does not pass these tests with `CONFIG_CAN_FD_MODE=y`, then the S32 CANXL driver needs to be fixed. 

From the looks of it, the driver needs to treat `CONFIG_CAN_NXP_S32_RX_FIFO=n` just like it treats `CONFIG_CAN_FD_MODE=n` by excluding all CAN FD functionality if the Rx FIFO is used.

Reply:
This is limitation of CANXL driver after removing CAN FD filtering.
Actually, it still works with `CONFIG_CAN_NXP_S32_RX_FIFO=n` and `CONFIG_CAN_FD_MODE=y` but it just supports to receive CAN classic frame. So it does not pass these tests and I disable CAN FD to skip these.

Reply:
The CAN FD test are skipped if the driver does not have the `CAN_MODE_FD` capability. This capability is reported as supported by the NXP S32 CANXL driver even for `CONFIG_CAN_NXP_S32_RX_FIFO=n`. This does not look right.

Reply:
yes, that's right. CANXL driver is supporting CAN FD capability even for `CONFIG_CAN_NXP_S32_RX_FIFO=n`. I will open other PR to removed CAN FD support for non-RX_FIFO.

Reply:
Just fix it in this PR?

Reply:
No, please leave the guards in. They are correct even if this test will not be built without CAN FD support.